src/gui/src/mainWindow.cpp

```diff
@@ -291,20 +293,28 @@ const std::string MainWindow::addToolbarButton(const std::string& name,
                                                const QString& script,
                                                bool echo)
 {
+  // ensure key is unique
+  std::string key;
+  if (name.empty()) {
+    int key_idx = 0;
+    do {
+      // default to "buttonX" naming
+      key = "button" + std::to_string(key_idx);
+      key_idx++;
+    } while (buttons_.count(key) != 0);
```

Comment:
this is n^2 but I guess n will be small.

Reply:
I'll merge it in for now, since unless you have a very large monitor, `n` should be pretty small. Do you have a better method in mind for generating unique names?

Reply:
**static** int key_idx = 0;   is the simplest though if you delete a button the index won't be reused (not a big deal)